gnome: Try to reduce output closure

jtojnar opened the pull request:
###### Motivation for this change
Try to do something about https://github.com/NixOS/nixpkgs/issues/141521

Closes: https://github.com/NixOS/nixpkgs/pull/114486

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

jtojnar committed:
gnome.nautilus: add dev output

So that propagatedBuildInputs do not pull gnome-autoar.dev into the out closure.

jtojnar committed:
libgxps: add dev output

So that propagatedBuildInputs do not pull libarchive.dev into the out closure.

jtojnar committed:
adwaita-qt: add dev output

So that gtk3.dev is not included in the closure through CMake files.

jtojnar committed:
python3.pkgs.dogtail: add dev output

So that propagatedBuildInputs do not pull gtk3.dev into the out closure.

jtojnar committed:
python3.pkgs.dogtail: add dev output

So that propagatedBuildInputs do not pull gupnp.dev into the out closure.

jtojnar committed:
gnome.gnome-user-share: fix apache module path

Also potentially reduces the number of dev dependencies in closure.

jtojnar committed:
mate.mate-user-share: fix apache module path

Also potentially reduces the number of dev dependencies in closure.

jtojnar committed:
gnome.gnome-music: move python dependencies to pythonPath

It is deprecated but otherwise wrapPython adds everything that is propagated, including dev outputs, to program_PATH, which ends up in the wrapper.

jtojnar committed:
gnome.gnome-music: fix double wrapping

jtojnar committed:
mod_dnssd: run {pre,post}Install hooks

jtojnar committed:
mod_dnssd: strip debugging symbols

Otherwise gnome-user-share and thus the whole GNOME will pull in apacheHttpd.dev & co.

jtojnar committed:
gnome.gnome-tweaks: move python dependencies to pythonPath

It is deprecated but otherwise wrapPython adds everything that is propagated, including dev outputs, to program_PATH, which ends up in the wrapper.

jtojnar committed:
orca: move python dependencies to pythonPath

It is deprecated but otherwise wrapPython adds everything that is propagated, including dev outputs, to program_PATH, which ends up in the wrapper.

roberth commented:
This seems to be about _output_ closure size, not _build_ closure size. Reading through the diff, the graph structure of the derivation closure seems to stay the same.

vcunat commented:
If I look right, now it's only 6 MiB to get under the limit again.

jtojnar commented:
I think I can resolve the gstreamer issue as well.

jtojnar committed:
gst_all_1.gst-plugins-bad: disable asserts

`lib/gstreamer-1.0/libgstdvbsubenc.so` and `lib/libgstcodecparsers-1.0.so` include `${gstreamer.dev}/include/gstreamer-1.0/gst/base/gstbitwriter.h` in their .rodata section, increasing the runtime closure unnecessarily. This happens because `gstbitwriter.h` header uses GLib’s `g_assert` macro, which wants to print `__FILE__` on assertion failure.

The assertions can and should be disabled on stable releases (https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/7b8d74c15bb817c65c06c63580dae3633a914841) so let’s do that.

jtojnar committed:
gst_all_1.gst-plugins-good: disable asserts

`lib/gstreamer-1.0/libgstrtpmanager.so` includes `${gstreamer.dev}/include/gstreamer-1.0/gst/base/gstbitwriter.h` in their .rodata section, increasing the runtime closure unnecessarily. This happens because `gstbitwriter.h` header uses GLib’s `g_assert` macro, which wants to print `__FILE__` on assertion failure.

The assertions can and should be disabled on stable releases (https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/ce64f0e0e0d4e692c8d99f9744d38b9676b3d720) so let’s do that.
